### PR TITLE
Require Ruby 2.4+ + update dependencies to latest versions

### PR DESCRIPTION
**Pull request by ekohl**
This updates all dependencies to their current versions and drops support for Ruby 2.3 and older.

### CODE REVIEW

**Comment by ekohl** [comment]
@ekohl interesting that in Ruby 2.7 the order is the other way around again.

**Review by bastelfreak — APPROVED** [review]

**Comment by ekohl** [comment]
Should we leave out 2.7 for now to make it green?

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,4 @@ source "https://rubygems.org"
 # Specify your gem's dependencies in librarian.gemspec
 gemspec
 
-gem "fakefs"
```
Comment (ghoneycutt) [inline Gemfile:6]: why is this removed?

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@
 pkg/*
 tmp
 vendor
+Gemfile.lock
```
Comment (ghoneycutt) [inline .gitignore:6]: 👍

**Comment by ghoneycutt** [comment]
We should only test ruby versions that correspond to the latest releases (2.4.9 and 2.5.7) of the supported Puppet platforms (5 and 6)

https://puppet.com/docs/puppet/5.5/about_agent.html
https://puppet.com/docs/puppet/latest/about_agent.html

**Comment by ghoneycutt** [comment]
Awesome work @ekohl !

File: Gemfile (hunk 1 above)
Comment (ekohl) [reply Gemfile:6]: This is already listed in the development dependencies in the gemspec.

**Comment by ekohl** [comment]
> We should only test ruby versions that correspond to the latest releases (2.4.9 and 2.5.7) of the supported Puppet platforms (5 and 6)

I think this is not that important and it's good to be future proof by testing newer Rubies. However, I'll leave off 2.7 for now.

**Review by bastelfreak — APPROVED** [review]

**Review by ehelms — APPROVED** [review]

**Comment by ghoneycutt** [comment]
@ekohl Testing future versions is great, though suggest adding those to the `allowed_failures` option, so they don't keep anyone from merging. We do that here - https://github.com/sensu/sensu-puppet/blob/master/.travis.yml#L237 - in order to test future versions of dependencies as well as nightly builds of the software being managed.

**Review by ghoneycutt — APPROVED** [review]
Great work!

**Comment by ekohl** [comment]
I'm personally not a fan of allowed_failures since you do use CI resources but often don't actually look at them and act on it.
